plugins/test/validate.py

```diff
@@ -0,0 +1,164 @@
+      documentation that is represented by the value of C(engine) option.
+    - The plugin configuration option can be either passed as I(key=value) pairs within test plugin
+      or set as environment variables.
+    - The precedence the C(validate) plugin configurable option is the variable passed within test plugin
+      as I(key=value) pairs followed by task variables followed by environment variables.
+"""
+
+EXAMPLES = r"""
+- name: set facts for data and criteria
+  set_fact:
+    data: "{{ lookup('file', './validate/data/show_interfaces_iosxr.json')}}"
+    criteria: "{{ lookup('file', './validate/criteria/jsonschema/show_interfaces_iosxr.json')}}"
+
+- name: validate data in json format using jsonschema with test plugin
+  ansible.builtin.set_fact:
+    is_data_valid: "{{ data is ansible.utils.validate(engine='ansible.utils.jsonschema', criteria=criteria, draft='draft7') }}"
+"""
+
+RETURN = """
+  _raw:
+    description:
+      - If data is valid return C(true)
+      - If data is invalid return C(false)
+"""
+
+from copy import deepcopy
+
+from ansible.errors import AnsibleError
+from ansible.module_utils._text import to_native
+from ansible_collections.ansible.utils.plugins.module_utils.validate.base import (
+    load_validator,
+)
+from ansible_collections.ansible.utils.plugins.module_utils.common.utils import (
+    to_list,
+)
+from ansible_collections.ansible.utils.plugins.module_utils.common.argspec_validate import (
+    check_argspec,
+)
+
+try:
+    import yaml
+
+    try:
+        from yaml import CSafeLoader as SafeLoader
+    except ImportError:
+        from yaml import SafeLoader
+    HAS_YAML = True
+except ImportError:
+    HAS_YAML = False
+
+ARGSPEC_CONDITIONALS = {}
+
+
+def validate(*args, **kwargs):
+    if not len(args):
+        raise AnsibleError(
+            "Missing either 'data' value in test plugin input,"
+            "refer ansible.utils.validate test plugin documentation for detials"
+        )
+
+    params = {"data": args[0]}
+    for item in ["engine", "criteria"]:
+        if kwargs.get(item):
+            params.update({item: kwargs[item]})
+
+    argspec = deepcopy(DOCUMENTATION)
+    argspec_obj = yaml.load(argspec, SafeLoader)
+
+    argspec_result, updated_params = check_argspec(
+        yaml.dump(argspec_obj),
+        "test",
+        schema_conditionals=ARGSPEC_CONDITIONALS,
+        **params
+    )
+    if argspec_result.get("failed"):
+        raise AnsibleError(
+            "%s with errors: %s"
+            % (argspec_result.get("msg"), argspec_result.get("errors"))
+        )
+
+    validator_engine, validator_result = load_validator(
+        engine=updated_params["engine"],
+        data=updated_params["data"],
+        criteria=updated_params["criteria"],
+        kwargs=kwargs,
+    )
+    if validator_result.get("failed"):
+        raise AnsibleError(
+            "validate lookup plugin failed with errors: %s"
```

Comment:
I noticed you and I format strings differently... should we agree on one approach?

Reply:
I used this format as it results in shorter strings but I do agree that the format you used is more readable hence reverted the code to use string formatting as you do at applicable places.